src/core/core.js

```diff
@@ -746,4 +748,21 @@ export default class Core {
     }
     return urls;
   }
+
+  getSessionId() {
+    if (this.storage.get(StorageKeys.SESSIONS_OPT_IN).value) {
+      try {
+        let sessionId = window.sessionStorage.getItem('sessionId');
+        if (!sessionId) {
+          sessionId = uuid();
+          window.sessionStorage.setItem('sessionId', sessionId);
```

Comment:
nit: I generally try to avoid putting setters inside of a getters because that means the getter has side effects. What do you think of renaming to `getOrSetupSessionId` to try and avoid this confusion?

Reply:
I agree that the name of the function could be better. getSessionId() is definitely misleading since it's not just a getter.